metabase/metabase
# Add focus trap to modals to prevent background keyboard navigation

Closes #30489
### Description
  
  Uses the Mantine `FocusTrap` component to ensure that keyboard navigation stays within the window modal. While the issue only refers to this issue in the context of the ObjectDetail modal when viewing an item in a Question table, this issue also occurs in these modals: 
  

- Create a new collection
- Move a question/dashboard/etc
- "About Metabase"
- Add question to a dashboard
- Pick action to add to dashboard

<details>
<summary> Old content for context. I've removed the breakpoint from `CloseButton` which avoids the problem. </summary>

**The main issue is viewing row details within a model.** On small screens, the breakpoint on the CSS of the `CloseButton` causes the button to use `display: none`, causing the button to leave the accessibility tree. This behavior is fine, but for some reason, it causes Mantine's focus trap to jump away from the modal and inserts itself back into the body, causing the behavior that we're trying to avoid. I put in some CSS here https://github.com/metabase/metabase/blob/e2a205b0c9d70353bf60c16b14ab064975b20133/frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.styled.tsx#L58-L63

and I absolutely hate this, but it keeps the button in the accessibility tree and allows us to keep the focus within the modal. This fix would be a one-off, but I hope there's a better way to solve this. There doesn't seem to be any guidance from Mantine or its github issues so I'm not sure if there would be a better way to solve this.

</details>



### How to verify
  
  To replicate the original issue:  
  

1. Go to a model
2. Click the "Expand" button next to any row in the model
3. Press `Tab` multiple times, you will see that the focus can still go behind the modal, and clicking `Enter` will click the buttons.  

  
With this PR, the focus should remain within the modal, and pressing `Tab` will only cycle through the elements within the modal.

### Checklist
